RestaurantsNearMe/Presentation/Components/Toast.swift

```diff
@@ -9,24 +9,26 @@ import SwiftUI
 
 struct Toast: View {
   let message: String
-  @Environment(\.deviceOrientation) var deviceOrientation
+  var parentSize: CGSize
 
   var body: some View {
-    RoundedRectangle(cornerRadius: 10.0)
-      .strokeBorder()
-      .fill(Color.clear)
-      .containerRelativeFrame(.horizontal) { layout, _ in
-        layout * UIConstants.toastWidthRatio
-      }
-      .containerRelativeFrame(.vertical) { layout, _ in
-        deviceOrientation == .portrait
-          ? layout * UIConstants.toastHeightRatioPortrait
-          : layout * UIConstants.toastHeightRatioLandscape
-      }.overlay {
-        Text(message)
-          .font(.largeTitle)
-          .bold()
-      }
+      RoundedRectangle(cornerRadius: 10.0)
```

Comment:
In order to center this toast in a VStack, I had to use the GeometryReader at one level above this and pass it down, not sure if there's a better way to center items nested within a GeometryReader without doing .maxWidth/Height = .infinity

Reply:
That's odd, a `VStack` centres its content by default, do you have a rough sketch or something you can share?

Reply:
Sure (excuse my poor handwriting)
![IMG_0081](https://github.com/ColinEvans/RestaurantsNearMe/assets/9354994/8a7a8eff-bb44-4b43-8da3-bd8e23e95a36)

What was happening, was when the `GeometryReader` was just on one component within the VStack, I suppose it was given too much space by the `GeometryReader` and taking up a large portion of the screen?

So I ended up achieving what I wanted with:
```
 GeometryReader { proxy in
      ScrollView(.vertical){
        VStack {
          Text("Restaurants Near Me")
            .font(.title)
          
          if let errorToPresent {
            Toast(
              message: errorToPresent,
              parentSize: proxy.size
            )
          } else {
            ProgressView()
          }
        }
        .frame(
          maxWidth: .infinity,
          minHeight: proxy.size.height,
          maxHeight: .infinity
        )
      }.refreshable {
        viewModel.refresh()
      }
    }
  }
```
It matches what I was looking for, but style wise seems interesting to pass in the `Toast` size explicitly. Seems to be necessary though since I want to isolate the relative `Toast` sizing in the component itself.